packages/core/contracts/insured-bridge/ovm/OVM_OETH_BridgeDepositBox.sol

```diff
@@ -0,0 +1,81 @@
+     * @param _chainId L2 Chain identifier this deposit box is deployed on.
+     * @param _l1Weth Address of Weth on L1. Used to inform if the deposit should wrap ETH to WETH, if deposit is ETH.
+     * @param _l2Eth Address of ETH on L2. Used to inform the token that should be sent over the Optimism bridge.
+     * @param _l1EthWrapper Address of custom ETH wrapper on L1. Any ETH sent to this contract will be wrapped to WETH
+     *     and sent to the WETH Bridge Pool.
+     * @param _l1Weth Address of Weth on L1. Used to inform if the deposit should wrap ETH to WETH, if deposit is ETH.
+     * @param timerAddress Timer used to synchronize contract time in testing. Set to 0x000... in production.
+     */
+    constructor(
+        address _crossDomainAdmin,
+        uint64 _minimumBridgingDelay,
+        uint256 _chainId,
+        address _l1Weth,
+        address _l2Eth,
+        address _l1EthWrapper,
+        address timerAddress
+    ) OVM_BridgeDepositBox(_crossDomainAdmin, _minimumBridgingDelay, _chainId, _l1Weth, timerAddress) {
+        l2Eth = _l2Eth;
+        l1EthWrapper = _l1EthWrapper;
+    }
+
+    /**
+     * @notice Called by relayer (or any other EOA) to move a batch of funds from the deposit box, through the canonical
+     *     token bridge, to the L1 Withdraw box. Implementation is exactly the same as the standard OVM_BridgeDepositBox
+     * except constructed to work with Optimism ETH by first unwrapping WETH then bridging OETH. The target on L1 is
+     * not the bridgePool but to the l1EthWrapper that takes any ETH sent to it, wraps it and sends to the BridgePool.
+     * @dev The frequency that this function can be called is rate limited by the `minimumBridgingDelay` to prevent spam
+     *      on L1 as the finalization of a L2->L1 tx is quite expensive.
+     * @param l2Token L2 token to relay over the canonical bridge.
+     * @param l1Gas Unused by optimism, but included for potential forward compatibility considerations.
+     */
+    function bridgeTokens(address l2Token, uint32 l1Gas) public override nonReentrant() {
+        uint256 bridgeDepositBoxBalance = TokenLike(l2Token).balanceOf(address(this));
+        require(bridgeDepositBoxBalance > 0, "can't bridge zero tokens");
+        require(canBridge(l2Token), "non-whitelisted token or last bridge too recent");
+
+        whitelistedTokens[l2Token].lastBridgeTime = uint64(getCurrentTime());
+
+        address bridgePool = whitelistedTokens[l2Token].l1BridgePool;
+
+        // If the L2 token is L2WETH then, to work with the canonical optimism bridge, we first unwrap it to ETH then
+        // bridge ETH over the canonical bridge. On L1 the l1EthWrapper will re-wrap the ETH to WETH and send it to
+        // the WETH bridge pool.
+        if (whitelistedTokens[l2Token].l1Token == l1Weth) {
+            WETH9Like(l2Token).withdraw(bridgeDepositBoxBalance);
+            l2Token = l2Eth;
```

Comment:
This block has I think some redundant code. For example, if we're going to assume that IF `whitelistedTokens[l2Token].l1Token == l1Weth` means that we're dealing with the WETH token, then why do we need to set `l2Token = l2Eth`? 

What if we changed the if statement to: `if (l2Token == l2Eth && whitelistedTokens[l2Token].l1Token == l1Weth)`, or alternatively just checking `if (l2Token == l2Eth)`

Reply:
Or, since setting a whitelisted L2 token's `l1Token = l1weth.address` is what we're assuming officially configures an L2 token as the WETH token, then we should just remove the line `l2Token = l2Eth`. 

Phrased differently: We should already be assuming that any `whitelistedTokens[l2Token]` whose `l1Token = l1WethAddress` is the L2 WETH token.

Reply:
this is not redundant code as the l2Token is set to l2Weth, not l2Eth. we still need to enable users to send L2Weth and therefore we store the l2Token as the weth deployed on L2.

This also makes the implementation behave the same as other chains (arbitrum) wherein the l2Token(for the weth pool) is set to WETH. 

This extra logic is to get the canonical bridge to take _eth_ over. 

in your other message you said setting this:
https://github.com/UMAprotocol/protocol/pull/3611#discussion_r750524744 That wont work as we need to send _eth_ over the bridge, not weth. 

Put differently we need to do two things on the L2 side:
1) we need to accommodate the ability to have users deposit WETH or ETH on L2.
2) we need to bridge over the canonical bridge using ETH.

to accommodate this we need to store both of these addresses.

Reply:
Ahh this is really tricky. I think you should add a comment explaining the subtle difference between L2Weth and L2Eth